From e8dbc14149e7b15054d90e6b5dfbc7e29a76e018 Mon Sep 17 00:00:00 2001
From: Pantelis Antoniou <panto@antoniou-consulting.com>
Date: Wed, 13 Mar 2013 16:55:15 +0200
Subject: [PATCH 124/177] cape-tester: Add pinmux helper for drvvbus gpio

We need this so that we can control the power to the USB host port.
---
 firmware/capes/cape-bone-tester-00A0.dts | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/firmware/capes/cape-bone-tester-00A0.dts b/firmware/capes/cape-bone-tester-00A0.dts
index 4cc3575..d5b7607 100644
--- a/firmware/capes/cape-bone-tester-00A0.dts
+++ b/firmware/capes/cape-bone-tester-00A0.dts
@@ -198,6 +198,17 @@
 					0x194 0x37 	/* mcasp0_fsx.gpio3_15, OMAP_MUX_MODE7 | INPUT_PULLUP */
 				>;
 			};
+
+			bone_default_drvvbus_pins: pinmux_bone_default_drvvbus_pins {
+				pinctrl-single,pins = <
+					0x234 0x20 /* USB1_DRVVBUS musb  */
+				>;
+			};
+			bone_gpio_drvvbus_pins: pinmux_bone_gpio_drvvbus_pins {
+				pinctrl-single,pins = <
+					0x234 0x07 /* USB1_DRVVBUS.gpio3_13  OUTPUT  */
+				>;
+			};
 		};
 	};
 
@@ -418,6 +429,15 @@
 				pinctrl-1 = <&bone_tester_input_gpio_pins>;
 				pinctrl-2 = <&bone_tester_output_gpio_pins>;
 			};
+
+			drvvbus_pinmux_helper {
+				compatible = "bone-pinmux-helper";
+				status = "okay";
+
+				pinctrl-names = "default", "gpio";
+				pinctrl-0 = <&bone_default_drvvbus_pins>;
+				pinctrl-1 = <&bone_gpio_drvvbus_pins>;
+			};
 		};
 
 	};
-- 
1.8.1.4

